contracts/ERC1155.sol

```diff
@@ -22,16 +23,16 @@ contract ERC1155 is IERC1155, ERC165
 
 /////////////////////////////////////////// ERC165 //////////////////////////////////////////////
 
-    function supportsInterface(bytes4 _interfaceId)
-    external
-    view
-    returns (bool) {
-         // ToDo recalc interface.
-         if (_interfaceId == 0) {
-            return true;
-         }
-
-         return false;
+    constructor() internal {
+        _registerInterface(
+            // this.supportsInterface.selector ^ // <- include this?
```

Comment:
I don't think [ERC165](https://eips.ethereum.org/EIPS/eip-165) specified what to do in case interfaces have an inheritance structure.